pkg/pillar/diskmetrics/usage.go

```diff
@@ -201,3 +203,48 @@ func PersistUsageStat(log *base.LogObject) (*types.UsageStat, error) {
 	}
 	return usageStat, nil
 }
+
+// FoundLargeFiles is returned by FindLargeFiles
+type FoundLargeFiles struct {
+	Path string
+	Size int64
+}
+
+// FindLargeFiles walks a directory and reportes all large files unless
+// they are in an excluded directory
+func FindLargeFiles(root string, limit int64, excludePaths []string) ([]FoundLargeFiles, error) {
+	var list []FoundLargeFiles
+	walkErr := filepath.WalkDir(filepath.Clean(root), func(path string, di fs.DirEntry, err error) error {
+		// We skip all directories
+		if di.IsDir() {
+			return nil
+		}
+
+		// if there is any problem with path we stop
+		if err != nil {
+			return err
+		}
+
+		// Part of excludePath?
+		for _, ex := range excludePaths {
+			if filepath.HasPrefix(path, ex) {
```

Comment:
I don't think we have any problem with symlinks or anything like that now, but maybe it would make sense to normalize the paths before comparing.